### PR TITLE
Add PSK in libspdm_start_session(), remove PSK in context provision

### PR DESCRIPTION
**Pull request by jyao1**
Fix https://github.com/DMTF/libspdm/issues/1878

### CODE REVIEW

File: library/spdm_requester_lib/libspdm_req_communication.c [hunk 1]
```diff
@@ -53,6 +55,7 @@ libspdm_return_t libspdm_start_session(void *spdm_context, bool use_psk,
 
     if (!use_psk) {
     #if LIBSPDM_ENABLE_CAPABILITY_KEY_EX_CAP
+        LIBSPDM_ASSERT((psk_hint == NULL) && (psk_hint_size == 0));
```
Comment (steven-bellock) [inline library/spdm_requester_lib/libspdm_req_communication.c:58]: These parameters can just be ignored if it's not a PSK session. No need to force Integrator to set their values.

**Review by steven-bellock — CHANGES_REQUESTED** [review]
If `PskHint` is a Requester parameter then `OpaqueData` should be as well. But that can be a separate pull request.

File: library/spdm_requester_lib/libspdm_req_communication.c (hunk 1 above)
Comment (jyao1) [reply library/spdm_requester_lib/libspdm_req_communication.c:58]: why?
There is no meaning to input PSK_HINT for DHE session. I would treat it as user mistake.

**Comment by jyao1** [comment]
> If PskHint is a Requester parameter then OpaqueData should be as well. But that can be a separate pull request.

This issue/PR only handles PskHint.
Please do file separate one for PSK or KeyExchange.

File: library/spdm_requester_lib/libspdm_req_communication.c (hunk 1 above)
Comment (jyao1) [reply library/spdm_requester_lib/libspdm_req_communication.c:58]: No.  libspdm does not evaluate psk_hint or psk_hint_size if the session is not PSK.

My intention is:
1. For DHE session, PSK_HINT must be NULL.
2. For PSK session, PSK_HINT could be NULL or non-NULL.

I am not sure why we need allow DHE with PSK_HINT non-NULL, and ignore them.
What is the use case we want to support?

Comment (steven-bellock) [reply library/spdm_requester_lib/libspdm_req_communication.c:58]: >What is the use case we want to support?

No use case. Just in the documentation I would like it to say
> If `use_psk` is false then the `psk_hint` and `psk_hint_size` parameters are ignored.

Comment (jyao1) [reply library/spdm_requester_lib/libspdm_req_communication.c]: updated.

File: include/library/spdm_requester_lib.h [hunk 2]
```diff
@@ -285,6 +285,8 @@ libspdm_return_t libspdm_get_measurement_ex(void *spdm_context, const uint32_t *
  * @retval RETURN_SECURITY_VIOLATION    Any verification fails.
  **/
 libspdm_return_t libspdm_start_session(void *spdm_context, bool use_psk,
+                                       const void *psk_hint,
```
Comment (steven-bellock) [inline include/library/spdm_requester_lib.h:288]: These parameters need to be added to the Doxygen comment above. I can update the API documentation.

File: include/library/spdm_requester_lib.h [hunk 3]
```diff
@@ -328,6 +330,8 @@ libspdm_return_t libspdm_start_session(void *spdm_context, bool use_psk,
  * @retval RETURN_SECURITY_VIOLATION    Any verification fails.
  **/
 libspdm_return_t libspdm_start_session_ex(void *spdm_context, bool use_psk,
+                                          const void *psk_hint,
```
Comment (steven-bellock) [inline include/library/spdm_requester_lib.h:333]: These parameters need to be added to the Doxygen comment above. I can update the API documentation.

**Review by steven-bellock — CHANGES_REQUESTED** [review]

File: include/library/spdm_requester_lib.h (hunk 2 above)
Comment (jyao1) [reply include/library/spdm_requester_lib.h:288]: Added.

File: include/library/spdm_requester_lib.h (hunk 3 above)
Comment (jyao1) [reply include/library/spdm_requester_lib.h:333]: Added.

**Comment by jyao1** [comment]
I merge it to avoid conflict later. Feel free to file new issue, if you find anything need improvement.

**Pull request merged**